lib/Driver/UnixToolChains.cpp

```diff
@@ -320,6 +322,50 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
   return II;
 }
 
+
+ToolChain::InvocationInfo
+toolchains::GenericUnix::constructInvocation(const ArchiveJobAction &job,
+                               const JobContext &context) const {
+   assert(context.Output.getPrimaryOutputType() == file_types::TY_Image &&
+         "Invalid linker output type.");
+
+  ArgStringList Arguments;
+
+  // Configure the toolchain.
+  const char *AR = nullptr;
+  if (const Arg *A = context.Args.getLastArg(options::OPT_tools_directory)) {
+    StringRef toolchainPath(A->getValue());
+
+    // If there is an llvm-ar in the toolchain folder, use that instead.
+    if (auto toolchainAR =
+            llvm::sys::findProgramByName("llvm-ar", {toolchainPath})) {
+      AR = context.Args.MakeArgString(toolchainAR.get());
+    }
+  }
+
+  if (AR == nullptr) {
+    if (auto pathAR = llvm::sys::findProgramByName("ar", None))
+      AR = context.Args.MakeArgString(pathAR.get());
+  }
+
+  assert(AR &&
+         "neither ar nor llvm-ar was not found in the toolchain directory or system path.");
+
+  Arguments.push_back("crs");
+
+  Arguments.push_back(
+      context.Args.MakeArgString(context.Output.getPrimaryOutputFilename()));
+
+  addPrimaryInputsOfType(Arguments, context.Inputs, context.Args,
+                         file_types::TY_Object);
+  addInputsOfType(Arguments, context.InputActions, file_types::TY_Object);
+
+  InvocationInfo II{AR, Arguments};
+  II.allowsResponseFiles = true;
```

Comment:
mmm I don't think that the traditional archiver permits response files.  This should be equivalent to whether `llvm-ar` is in use.